macros/core/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Implemntation of the procedural macros exposed via the `gprogram-framework-macros` crate.
+
+use proc_macro2::TokenStream as TokenStream2;
+
+mod handlers;
+
+const COMMAND_ENUM_NAME: &str = "Commands";
+const COMMAND_RESPONSES_ENUM_NAME: &str = "CommandResponses";
+const QUERY_ENUM_NAME: &str = "Queries";
+const QUERY_RESPONSES_ENUM_NAME: &str = "QueryResponses";
+
+pub fn command_handlers_core(mod_tokens: TokenStream2) -> TokenStream2 {
+    handlers::generate(mod_tokens, COMMAND_ENUM_NAME, COMMAND_RESPONSES_ENUM_NAME)
+}
+
+pub fn query_handlers_core(mod_tokens: TokenStream2) -> TokenStream2 {
+    handlers::generate(mod_tokens, QUERY_ENUM_NAME, QUERY_RESPONSES_ENUM_NAME)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use quote::quote;
+
+    #[test]
+    fn command_handlers_core_works() {
+        let input = quote! {
+            mod commands {
+                use super::*;
+
+                struct SomeStruct {}
+
+                fn do_this(p: SomeStruct) {}
+            }
+        };
+        let expected = quote! {
+            mod commands {
+                extern crate parity_scale_codec as commands_scale_codec;
+                extern crate scale_info as commands_scale_info;
+
+                #[derive(commands_scale_codec::Encode, commands_scale_codec::Decode, commands_scale_info::TypeInfo)]
```

Comment:
Too wordy too
```suggestion
                #[derive(Encode, Decode, TypeInfo)]
```

Reply:
I've done it on purpose for the generated code in order to avoid possible name conflicts